src/Form/matching.js

```diff
@@ -500,24 +504,150 @@ class Matching {
     }
 }
 
+/**
+ *  @returns {SupportedTypes}
+ */
+function getInputType (input) {
+    const attr = input.getAttribute(ATTR_INPUT_TYPE)
+    if (isValidSupportedType(attr)) {
+        return attr
+    }
+    return 'unknown'
+}
+
+/**
+ * Retrieves the main type
+ * @param {SupportedTypes} type
+ * @returns {SupportedMainTypes}
+ */
+function getMainTypeFromType (type) {
+    const mainType = type.split('.')[0]
+    switch (mainType) {
+    case 'credentials':
+    case 'creditCard':
+    case 'identities':
+        return mainType
+    }
+    return 'unknown'
+}
+
 /**
  * Retrieves the input main type
  * @param {HTMLInputElement} input
- * @returns {SupportedMainTypes | string}
+ * @returns {SupportedMainTypes}
  */
 const getInputMainType = (input) =>
-    input.getAttribute(ATTR_INPUT_TYPE)?.split('.')[0] ||
-    'unknown'
+    getMainTypeFromType(getInputType(input))
+
+/** @typedef {supportedIdentitiesSubtypes[number]} SupportedIdentitiesSubTypes */
+const supportedIdentitiesSubtypes = /** @type {const} */ ([
+    'emailAddress',
+    'cardName',
+    'firstName',
+    'middleName',
+    'lastName',
+    'fullName',
+    'phone',
+    'addressStreet',
+    'addressStreet2',
+    'addressCity',
+    'addressProvince',
+    'addressPostalCode',
+    'addressCountryCode',
+    'birthdayDay',
+    'birthdayMonth',
+    'birthdayYear'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedIdentitiesSubTypes}
+ */
+function isValidIdentitiesSubtype (supportedType) {
+    return supportedIdentitiesSubtypes.includes(supportedType)
+}
+
+/** @typedef {supportedCreditCardSubtypes[number]} SupportedCreditCardSubTypes */
+const supportedCreditCardSubtypes = /** @type {const} */ ([
+    'cardName',
+    'cardNumber',
+    'cardSecurityCode',
+    'expirationMonth',
+    'expirationYear',
+    'expiration'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedCreditCardSubTypes}
+ */
+function isValidCreditCardSubtype (supportedType) {
+    return supportedCreditCardSubtypes.includes(supportedType)
+}
+
+/** @typedef {supportedCredentialsSubtypes[number]} SupportedCredentialsSubTypes */
+const supportedCredentialsSubtypes = /** @type {const} */ ([
+    'password',
+    'username'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedCredentialsSubTypes}
+ */
+function isValidCredentialsSubtype (supportedType) {
+    return supportedCredentialsSubtypes.includes(supportedType)
+}
+
+/** @typedef {SupportedIdentitiesSubTypes | SupportedCreditCardSubTypes | SupportedCredentialsSubTypes} SupportedSubTypes */
+
+/** @typedef {`identities.${SupportedIdentitiesSubTypes}` | `creditCard.${SupportedCreditCardSubTypes}` | `credentials.${SupportedCredentialsSubTypes}` | 'unknown'} SupportedTypes */
+
```

Comment:
Should this be right above the `const` definition (basically remove this empty line)?